airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/core/config/ConnectorConfiguration.kt

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.core.config
+
+import com.fasterxml.jackson.databind.JsonNode
+import io.airbyte.commons.json.Jsons
+import java.util.Optional
+
+/** Interface that defines a typed connector configuration. */
+interface ConnectorConfiguration {
+    fun getDefaultNamespace(): Optional<String>
+
+    fun getRawNamespace(): Optional<String>
```

Comment:
Do we want to specialize this by source and destination? getRawNamespace is a destinations-only concept AFAIK.

Reply:
@postamar I would rather do the opposite:  just have one method called `getNamespace()` or something that is used by both, as there is no reason to have two different ones.  For now, this is purely to help us with converting code so we know what maps to what.